src/pull_request.ts

```diff
@@ -53,19 +47,76 @@ export async function handlePullRequest() {
   let overviewComment = existingComments.find((comment) =>
     comment.body?.includes(OVERVIEW_MESSAGE_SIGNATURE)
   );
+  const isIncrementalReview = !!overviewComment;
+
+  // Maybe fetch review comments
+  const reviewComments = isIncrementalReview
+    ? (
+        await octokit.rest.pulls.listReviewComments({
+          ...context.repo,
+          pull_number: pull_request.number,
+        })
+      ).data
+    : [];
+
+  // Get modified files
+  const { data: files } = await octokit.rest.pulls.listFiles({
+    ...context.repo,
+    pull_number: pull_request.number,
+  });
+  let filesToReview = files.map((file) => parseFileDiff(file, reviewComments));
+  info(`successfully fetched file diffs`);
+
+  let commitsReviewed: string[] = [];
   if (overviewComment) {
+    info(`running incremental review`);
+    try {
+      const payload = JSON.parse(
+        overviewComment.body
+          ?.split(PAYLOAD_TAG_OPEN)[1]
+          .split(PAYLOAD_TAG_CLOSE)[0] || "{}"
+      );
+      commitsReviewed = payload.commits;
+    } catch (error) {
+      warning(`error parsing overview payload: ${error}`);
+    }
```

Comment:
Consider adding type safety to the payload parsing by defining an interface and validating the parsed data. This would prevent runtime errors from malformed data:
```typescript
interface OverviewPayload {
  commits: string[];
}

function isValidPayload(data: unknown): data is OverviewPayload {
  return (
    typeof data === 'object' &&
    data !== null &&
    Array.isArray((data as OverviewPayload).commits) &&
    (data as OverviewPayload).commits.every(c => typeof c === 'string')
  );
}

try {
  const parsed = JSON.parse(/* ... */);
  if (!isValidPayload(parsed)) {
    warning('Invalid payload structure');
    commitsReviewed = [];
  } else {
    commitsReviewed = parsed.commits;
  }
}
```